torch/_inductor/ir.py

```diff
@@ -175,6 +177,10 @@
 ]
 
 
+PartitionInputType = Dict[str, Union["IRNode", sympy.Expr]]
```

Comment:
While these are helpful for type checking, for the more complicated classes you might just consider frozen data classes, which are more debuggable at runtime. although not necessarily saying we need it here.